src/app/_components/Event/EventFilter.js

```diff
@@ -0,0 +1,179 @@
+  });
+
+  function selectButtonIcon(index) {
+    if (activeIndex === index && semiorange.active) {
+      return <CheckMark key={`checkmark+${index}`} className="h-4 w-4 transition-all" />;
+    }
+    if (hoveredIndexes[index] && activeIndex !== index && (semiorange.hover || semiorange.focused)) {
+      return <Search key={`searchicon+${index}`} className="h-4 w-4 transition-all " />;
+    }
+    return null;
+  }
+
+  // Get 6 basic month names for filters, starting from current month
+  useEffect(() => {
+    const getNextMonths = () => {
+      const nextMonths = [];
+      for (let i = 0; i <= monthsAhead; i += 1) {
+        const nextMonthIndex = (currentMonth + i - 1) % 12;
+        const nextMonthName = monthNames[nextMonthIndex];
+        setActiveMonthNumber(monthNames[i] === currentMonth ? monthNames[i] : '');
+        nextMonths.push(nextMonthName);
+      }
+      return nextMonths;
+    };
+    const nextMonths = getNextMonths();
+    setMonths(nextMonths);
+    // eslint-disable-next-line
+  }, [currentMonth]);
+
+  // Fetch data and pass url params
+  const { data, error, isLoading, hasNextPage, fetchNextPage, isSuccess, isFetchingNextPage } = useInfiniteQuery({
+    queryFn: ({ month = activeMonthNumber, pageParam = '' }) => allEvents({ month, take: 6, lastCursor: pageParam }),
+    queryKey: ['event', activeMonthNumber],
+    getNextPageParam: lastPage => lastPage?.metaData.lastCursor,
+  });
+
+  // Get data on first render or on url open
+  useEffect(() => {
+    const monthFromQuery = searchParams.get('month');
+
+    if (monthFromQuery && monthFromQuery !== currentMonth) {
+      // If the month is defined in the query and is not equal to the current month
+      setActiveMonthNumber(monthFromQuery);
+      setActiveIndex(monthFromQuery - 1);
+      setActiveMonth(months.filter(item => item.index === parseInt(monthFromQuery, 10)));
+      allEvents({ month: monthFromQuery, take: '', lastCursor: '' });
+    } else if (!monthFromQuery || monthFromQuery === undefined) {
+      setActiveMonth(months[0]);
+      setActiveMonthNumber(currentMonth);
+      setActiveIndex(currentMonth - 1);
+      deleteParam();
+      addParam(currentMonth.toString());
+      allEvents({ month: currentMonth, take: '', lastCursor: '' });
+    }
+
+    // eslint-disable-next-line
+  }, [currentMonth]);
+
+  useEffect(() => {
+    // if the last element is in view and there is a next page, fetch the next page
+    if (inView && hasNextPage) {
+      fetchNextPage();
+    }
+    // eslint-disable-next-line
+  }, [hasNextPage, inView]);
+
+  // // Filter data based on selected month
+  const handleFilter = newMonth => {
+    setActiveIndex(newMonth.index);
+    setActiveMonth(newMonth);
+    const calcMonth = parseInt(newMonth.index + 1, 10);
+    setActiveMonthNumber(calcMonth);
+    deleteParam();
+    addParam(calcMonth);
+    allEvents({ month: calcMonth, take: '', lastCursor: '' });
```

Comment:
@Yivgha  same here for calling api, we shouldn't

Reply:
changed